os/board/common/crashdump.c

```diff
@@ -53,11 +69,138 @@
 /****************************************************************************
  * Private Variables
  ****************************************************************************/
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH)
+static char ramdump_partname[MTD_PARTNAME_LEN + 1];
+static uint32_t ramdump_partoffset = 0;
+static uint32_t ramdump_partsize = 0;
+#endif
+
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+static char coredump_partname[MTD_PARTNAME_LEN + 1];
+static uint32_t coredump_partoffset = 0;
+static uint32_t coredump_partsize = 0;
+#endif
 
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
 
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+static int coredump_to_flash(uint32_t cur_sp, void *tcb)
+{
+	uint32_t *ptr;
+	int32_t i;
+	uint32_t j;
+	uint32_t len = 0;
+	char pbuf[COREDUMP_BUF_SIZE];
+	int ret = -1;
+	int32_t coredump_size = 0;
+
+	sched_lock();
+
+	/* Add dummy coredump_size at the beginning */
+	len += sprintf(pbuf + len, "%08x\n", coredump_size);
+
+	/* space for '\0' character added by sprintf */
+	len += 1;
+
+	/* Process and format the stack to readable format */
+	len += sprintf(pbuf + len, "up_dumpstate:   size: %08x\n", \
+				((struct tcb_s *)tcb)->adj_stack_size);
+
+	ptr = (uint32_t *)cur_sp;
+	for (i = 0; i < 8; i++) {
+		len += sprintf(pbuf + len, "up_stackdump: %08x:", ptr);
+		for (j = 0; j < 8; j++) {
+			len += sprintf(pbuf + len, " %08x", ptr[j]);
+		}
+		len += sprintf(pbuf + len, "\n");
+		ptr += 8;
+	}
+
+	ptr = (uint32_t *)current_regs;
+	for (i = 0; i < 2; i++) {
+		len += sprintf(pbuf + len, "up_registerdump: R%d:", (i * 8));
+		for (j = 0; j < 8; j++) {
+			len += sprintf(pbuf + len, " %08x", ptr[j]);
+		}
+		len += sprintf(pbuf + len, "\n");
+		ptr += 8;
+	}
+
+	if (coredump_partsize < len) {
+		lldbg("Partition size 0x%x is less coredump size 0x%x\n", coredump_partsize, len);
+		goto err ;
+	}
+
+	/* calculate total formated coredump_size, without size field(10) bytes */
+	coredump_size = len - 10;
+
+	/* Add the actual crashlog_size at the beginning */
+	sprintf(pbuf, "%08x", coredump_size);
+
+	ret = crashdump_write_to_flash(coredump_partoffset, (uint32_t *)pbuf, len);
```

Comment:
How about using `up_xxxx` instead of `crashdump_write_to_flash`?
up_xx is the HAL API. I think we had better use the same usages.
Thapa is adding the tab of how to port RAMDUMP. In there, we will add the function list chip vendor should implement to use RAMDUMP. It says up_getc, up_putc and up_puts. This function also should be there. In this point of view, I leave this comment.

Reply:
If you have another chip-specific API which needs to use RAMDUMP, could you let me and Thapa know?

Reply:
done